src/debug-generator/debug-generator.c

```diff
@@ -142,6 +159,21 @@ static int generate_wants_symlinks(void) {
         return r;
 }
 
+static int install_debug_shell_dropin(const char *dir, const char *tty) {
+        int r;
+
+        r = write_drop_in_format(dir, "debug-shell.service", 50, "tty",
+                        "[Unit]\n"
+                        "Description=Early root shell on /dev/%s FOR DEBUGGING ONLY\n"
+                        "[Service]\n"
+                        "TTYPath=/dev/%s",
+                        tty, tty);
+        if (r < 0)
+                return log_debug_errno(r, "Failed to drop-in for debug-shell.service: %m");
```

Comment:
now you log twice about this error... please log either here or in the caller, but not in both and not in neither.

Reply:
(and i'd really opt for here, but in the caller is fine too)